lib/two_factor_authentication/hooks/two_factor_authenticatable.rb

```diff
@@ -11,7 +11,9 @@
 
   if user.respond_to?(:need_two_factor_authentication?) && !bypass_by_cookie
     if auth.session(options[:scope])[TwoFactorAuthentication::NEED_AUTHENTICATION] = user.need_two_factor_authentication?(auth.request)
-      user.send_two_factor_authentication_code
+      if !user.totp_enabled?
+        user.send_new_otp
+      end
     end
```

Comment:
The logic here is starting to get unwieldy. I personally don't think this logic belongs here. I think it fits better here: https://github.com/Houdini/two_factor_authentication/blob/6245f9d07cc1dec7e5ae2792db974683d91d4718/lib/two_factor_authentication/controllers/helpers.rb#L16

For now, I suppose this is fine, but this module needs to be refactored at some point.